light/client.go

```diff
@@ -956,6 +1008,26 @@ func (c *Client) getLightBlock(ctx context.Context, p provider.Provider, height
 	return l, err
 }
 
+func (c *Client) findIndexForWitness(ID types.NodeID) (int, bool) {
+	for i, w := range c.witnesses {
+		if w.ID() == string(ID) {
+			return i, true
+		}
+	}
+	return 0, false
+}
+
+// RemoveProviderByID removes a witness from the light client.
+func (c *Client) RemoveProviderByID(ID types.NodeID) error {
```

Comment:
We should probably add a check here to make sure we don't remove all witnesses and at least keep one witness?

Reply:
That check exists in the lower `removeWitness` function

Reply:
You are right, thanks!